src/controllers/authController.js

```diff
@@ -89,6 +89,52 @@ export default class AuthController {
     }
   }
 
+  /**
+     * @description Google and Facebook authentication method
+     * @static
+     * @param {Object} req
+     * @param {Object} res
+     * @returns {Object} User is successfully logged in.
+     * @returns {Object} User is scucessfilly created
+     * @memberof authController
+     */
+  static async oAuthLogin(req, res) {
+    try {
+      const profile = { ...req.user };
+      const method = profile.provider;
+      const { id } = profile;
+      const firstName = profile.name.givenName;
+      const lastName = profile.name.familyName;
+      const email = Array.isArray(profile.emails) ? profile.emails[0].value : '';
```

Comment:
Do we need to check specifically if it's an array or if the profile.emails are present? If the latter, ```profile.emails ? profile.emails[0].value : ''``` should do

Reply:
I change that.